mlflow/server/js/src/common/components/ErrorView.tsx

```diff
@@ -54,24 +46,24 @@ type ErrorViewImplProps = {
   statusCode: number;
   subMessage?: string;
   fallbackHomePageReactRoute?: string;
-  designSystemThemeApi?: any;
+  designSystemThemeApi?: DesignSystemThemeInterface;
```

Comment:
```suggestion
  designSystemThemeApi: DesignSystemThemeInterface;
```

it looks like the `WithDesignSystemThemeHoc` always passes in this prop, can we remove the optional?

Reply:
I think there are type mismatches. `WithDesignSystemThemeHoc` require `designSystemThemeApi: DesignSystemThemeInterface;`, but none of `ErrorView` components pass `designSystemThemeApi` prop. 

https://github.com/mlflow/mlflow/blob/15f7032004f4aecaaf848f7a334ace86f58820da/mlflow/server/js/src/experiment-tracking/components/RunNotFoundView.tsx#L19-L23

What is the best practice here? Is there a way to get `DesignSystemThemeInterface` as default value in `ErrorView.tsx`? or just revert the change in the file?

Reply:
ah interesting, it looks like the `WithDesignSystemThemeHoc` doesn't return the correct type for some reason. i think we can cast it like:

```tsx
export const ErrorView = WithDesignSystemThemeHoc(ErrorViewImpl) as React.ComponentType<
  Omit<ErrorViewImplProps, 'designSystemThemeApi'>
>;
```

at the end of the file. this should accurately portray what the types are

Reply:
Ignore the above, changing the props to this seems to work:

```tsx
type ErrorViewImplProps = DesignSystemHocProps & {
  statusCode: number;
  subMessage?: string;
  fallbackHomePageReactRoute?: string;
};
```

Reply:
makes sense. the fix is pushed